test/fileio.jl

```diff
@@ -60,4 +60,37 @@ let fn = tempname() * ".h5"
   @test all(keys(dat) .== ["b", "a", "G/z", "G/f"])
 end
 
+end # @testset track_order
+
+@static if HDF5.API.h5_get_libversion() >= v"1.10.5"
+
+@testset "h5f_get_dset_no_attrs_hint" begin
+  fn = tempname()
+  threshold = 300
+  h5open(
+    fn, "w";
+    libver_bounds = :latest,
+    meta_block_size = threshold
+  ) do f
+    HDF5.API.h5f_set_dset_no_attrs_hint(f, true)
+    @test HDF5.API.h5f_get_dset_no_attrs_hint(f)
+    f["test"] = 0x1
+    # We expect that with the hint, the offset will actually be 300
+    @test HDF5.API.h5d_get_offset(f["test"]) == threshold
+  end
+  @test filesize(fn) == threshold + 1
+  h5open(
+    fn, "w";
+    libver_bounds = :latest,
+    meta_block_size = threshold
+  ) do f
+    HDF5.API.h5f_set_dset_no_attrs_hint(f, false)
+    @test !HDF5.API.h5f_get_dset_no_attrs_hint(f)
+    f["test"] = 0x1
+    # We expect that with the hint, the offset will actually be 300
```

Comment:
This comment needs to change

Reply:
I first figured out what the minimum size would be, and picked a round number greater than that. 

https://twitter.com/markkitti/status/1530811272966586368?t=kyNK_e2Gwr87Xuv9mhYVog&s=19